agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -169,310 +95,328 @@ public CallReplacementTransformer() {}
-          String mname = iv.getMethodName(pgen);
-          Type[] args = iv.getArgumentTypes(pgen);
-          Type instance_type = iv.getReferenceType(pgen);
-          Type[] new_args = BCELUtil.insert_type(instance_type, args);
-          MethodDef orig = new MethodDef(cname + "." + mname, args);
-          if (debug_class)
-            System.out.printf("looking for %s in map %s%n", orig, current_method_map);
-          ReplacementClass call = current_method_map.get(orig);
-          if (call != null) {
-            call.cnt++;
-            String classname = call.method_class;
-            String methodname = mname;
-            debug_map.log(
-                "Replacing method %s.%s (%s) with %s.%s%n",
-                cname, mname, ArraysMDE.toString(args), classname, methodname);
-            il.append(
-                ifact.createInvoke(
-                    classname, methodname, iv.getReturnType(pgen), new_args, Const.INVOKESTATIC));
-          }
-          return (il);
-        }
+      debug_transform.log(
+          "getReplacementInstruction: EXIT exception getting method for instruction %s of %s.%s: %s%n",
+          invocation.toString(), mg.getClassName(), mg.getName(), e);
+      return null;
+    }
+    MethodDef call = getReplacementMethod(orig);
+    if (call == null) {
+      debug_transform.log(
+          "%s.%s: No replacement for %s%n", mg.getClassName(), mg.getName(), orig.toString());
+      debug_transform.log(
+          "getReplacementInstruction: EXIT %s.%s%n", mg.getClassName(), mg.getName());
+      return null;
+    }
 
+    debug_transform.log(
+        "%s.%s: Replacing method %s with %s%n",
+        mg.getClassName(), mg.getName(), orig.toString(), call);
+
+    InvokeInstruction instruction;
+
+    switch (inst.getOpcode()) {
+      case Const.INVOKEDYNAMIC:
+      case Const.INVOKEINTERFACE:
+      case Const.INVOKESPECIAL:
+      case Const.INVOKEVIRTUAL:
+        /*
+         * These are invocations where an implicit argument occurs in the call.
+         */
+        Type instanceType = invocation.getReferenceType(pgen);
+        Type[] arguments = BCELUtil.insert_type(instanceType, invocation.getArgumentTypes(pgen));
+        instruction =
+            ifact.createInvoke(
+                call.getClassname(),
+                call.getName(),
+                invocation.getReturnType(pgen),
+                arguments,
+                Const.INVOKESTATIC);
+        break;
+      case Const.INVOKESTATIC:
+        instruction =
+            ifact.createInvoke(
+                call.getClassname(),
+                call.getName(),
+                invocation.getReturnType(pgen),
+                invocation.getArgumentTypes(pgen),
+                Const.INVOKESTATIC);
+        break;
       default:
-        return (null);
+        debug_transform.log(
+            "getReplacementInstruction: EXIT wrong instruction type in %s.%s%n",
+            mg.getClassName(), mg.getName());
+        return null;
```

Comment:
I don't think you crash a `ClassFileTransformer`. I think you throw an `IllegalClassFormatException`, which turns out to the the same as having the `transform` method return `null`